core/src/main/java/novoda/lib/sqliteprovider/provider/SQLiteContentProviderImpl.java

```diff
@@ -63,10 +63,30 @@ protected SQLiteOpenHelper getDatabaseHelper(Context context) {
 
     @Override
     protected Uri insertInTransaction(Uri uri, ContentValues values) {
+        return insertInTransaction(uri, values, true);
+    }
+
+    @Override
+    protected int bulkInsertInTransaction(Uri uri, ContentValues[] values) {
+        int rowsCreated = 0;
+        for (ContentValues value : values) {
+            Uri insertUri = insertInTransaction(uri, value, false);
+            if (insertUri != null) {
+                rowsCreated++;
+            }
+            getWritableDatabase().yieldIfContendedSafely();
+        }
+        notifyUriChange(uri);
+        return rowsCreated;
+    }
+
+    private Uri insertInTransaction(Uri uri, ContentValues values, boolean shouldNotifyChange) {
         long rowId = helper.insert(uri, values);
-		Uri newUri = ContentUris.withAppendedId(uri, rowId);
-		notifyUriChange(newUri);
-		return newUri;
+        Uri newUri = ContentUris.withAppendedId(uri, rowId);
+        if (shouldNotifyChange) {
```

Comment:
could instead have two methods - insertWithoutNotification and insertWithNotification. Then we remove the branching and make it clearer what is happening?